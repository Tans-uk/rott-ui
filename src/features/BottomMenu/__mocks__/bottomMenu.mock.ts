import {ThemeConfig} from '../../../models'
import type {BottomMenuItemModel} from '../models'

export const bottomMenuListItemMock: BottomMenuItemModel<ThemeConfig>[] = [
  {
    testID: 'bottom-menu-item-1-test-id',
    title: 'Menu 1',
    icon: {
      name: 'ARROW_DOWN',
    },
    onPress: jest.fn(),
  },
  {
    testID: 'bottom-menu-item-2-test-id',
    title: 'Menu 2',
    icon: {
      name: 'ARROW_DOWN',
    },
    onPress: jest.fn(),
  },
]
