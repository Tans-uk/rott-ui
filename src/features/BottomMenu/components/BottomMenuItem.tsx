import {useContext, type FC} from 'react'

import {Linking} from 'react-native'

import {RottUiContext} from '../../../contexts'
import {ThemeConfig} from '../../../models'
import {Icon} from '../../Icon'
import {Image} from '../../Image'
import {Label} from '../../Label'
import {Pressable} from '../../Pressable'
import type {BottomMenuItemModel} from '../models'
import {BottomMenuStyles} from '../styles'

export const BottomMenuItem: FC<BottomMenuItemModel<ThemeConfig>> = ({
  testID,
  title,
  icon,
  image,
  containerStyle,
  onPress,
  url,
  phone,
}) => {
  const {colors} = useContext(RottUiContext)
  const isPhone = !!phone && !phone.isEmpty()

  return (
    <Pressable
      testID={testID}
      alignItemsCenter
      width={containerStyle ? 56 : undefined}
      style={containerStyle}
      onPress={async (event) => {
        if (process.env.NODE_ENV === 'development' && isPhone) return
        else if (isPhone) await Linking.openURL(`tel:${phone}`)
        else if (url && (await Linking.canOpenURL(url))) await Linking.openURL(url)
        else if (onPress) onPress(event)
      }}>
      {icon && !image && (
        <Icon
          {...icon}
          name={icon.name}
          width={icon?.width ? icon.width : 24}
          height={icon?.height ? icon.height : 24}
          variant={icon.variant ?? 'grey-900'}
          mode={icon?.mode}
          strokeWidth={icon?.strokeWidth ? icon?.strokeWidth : 2}
          marginBottom={8}
        />
      )}

      {image && !icon && (
        <Image
          {...image}
          width={image?.width ? image?.width : 24}
          height={image?.height ? image.height : 24}
          tintColor={image?.width && image?.height ? undefined : colors['grey-900']}
          marginBottom={8}
          resizeMode='contain'
          name={image.name}
        />
      )}

      {title && (
        <Label
          style={BottomMenuStyles().textStyle}
          variant='grey-900'
          fontSize='xs'
          fontWeight='bold'
          numberOfLines={1}>
          {title}
        </Label>
      )}
    </Pressable>
  )
}
