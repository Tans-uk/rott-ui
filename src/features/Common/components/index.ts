export * from './CommonItem'
export * from './CommonItemContainer'
