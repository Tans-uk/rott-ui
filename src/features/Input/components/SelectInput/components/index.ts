export * from './SelectInput'
export * from './SelectInputModalComponent'
