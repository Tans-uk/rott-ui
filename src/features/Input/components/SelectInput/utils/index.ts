export * from './listHeightNormalizer'
export * from './useModalHeightPercentageNormalizer'
export * from './sortListBySearchPriority'
