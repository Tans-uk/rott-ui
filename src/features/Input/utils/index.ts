export * from './formatByDateMode'
export * from './useInputStyleNormalizer'
