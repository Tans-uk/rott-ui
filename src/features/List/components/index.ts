export * from './List'
export * from './FavoriteSwipeComponent'
export * from './ListSkeletonItem'
