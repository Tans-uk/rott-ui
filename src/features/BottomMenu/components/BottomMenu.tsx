import {useContext, type FC} from 'react'

import {Platform} from 'react-native'

import {RottUiContext} from '../../../contexts'
import {useSafeArea} from '../../../hooks'
import {ThemeConfig} from '../../../models'
import {Content} from '../../Content'
import type {BottomMenuItemModel} from '../models'
import {BottomMenuItem} from './BottomMenuItem'

interface BottomMenuProps {
  menuItems: BottomMenuItemModel<ThemeConfig>[]
}

export const BottomMenu: FC<BottomMenuProps> = ({menuItems}) => {
  const {colors} = useContext(RottUiContext)
  const {bottom} = useSafeArea()

  let absoluteHeight = 64 + bottom // 56 dip yükseklik + 8 dip boşluk + bottom inset
  if (Platform.OS === 'ios' && bottom > 0) absoluteHeight -= 8 // ios'te navigation bar şeffaf ve boşluklu olduğu için 8 dip boşluğa gerek yoktur

  return (
    <Content
      testID='bottom-menu-container'
      row
      absolute
      noPadding
      bottom={0}
      paddingTop={12}
      paddingHorizontal={8}
      size='full'
      borderTopStartRadius={24}
      borderTopEndRadius={24}
      justifyContentSpaceAround
      backgroundColor={colors.secondary}
      height={absoluteHeight}>
      {menuItems.map((props) => (
        <BottomMenuItem key={`${props?.title ?? 'Main_'}`} {...props} />
      ))}
    </Content>
  )
}
