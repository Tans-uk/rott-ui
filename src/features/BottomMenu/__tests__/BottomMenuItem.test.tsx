import {bottomMenuListItemMock} from '../__mocks__'
import {render} from '../../../__tests__/utils/testUtils'
import {BottomMenuItem} from '../components'

describe('Component -> BottomMenuItem', () => {
  const testIDs = {
    bottomMenuItemTestId: 'bottom-menu-item-1-test-id',
  }

  it('bottom menu item componenti snapshot ile eşleşmeli', () => {
    const rendered = render(<BottomMenuItem {...bottomMenuListItemMock[0]} />)

    expect(rendered).toMatchSnapshot()
  })

  it('bottom menu item componenti ekranda gözükmeli', () => {
    const {bottomMenuItemTestId} = testIDs

    const {getByTestId} = render(<BottomMenuItem {...bottomMenuListItemMock[0]} />)

    const bottomMenuItem = getByTestId(bottomMenuItemTestId)

    expect(bottomMenuItem).toBeOnTheScreen()
  })

  it('verilen icon ekranda gözükmeli', () => {
    const {bottomMenuItemTestId} = testIDs

    const {getByTestId} = render(<BottomMenuItem {...bottomMenuListItemMock[0]} />)

    const bottomMenuItem = getByTestId(bottomMenuItemTestId)

    expect(bottomMenuItem).toBeOnTheScreen()
  })
})
