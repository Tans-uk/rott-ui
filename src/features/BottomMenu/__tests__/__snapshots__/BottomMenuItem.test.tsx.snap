// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component -> BottomMenuItem bottom menu item componenti snapshot ile eşleşmeli 1`] = `
<View
  flex={1}
  pointerEvents="box-none"
  style={
    {
      "flex": 1,
      "flexDirection": "column",
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    alignItemsCenter={true}
    animated={false}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "column",
      }
    }
    testID="bottom-menu-item-1-test-id"
  >
    <Text
      numberOfLines={1}
      style={
        {
          "color": "rgba(34, 63, 70, 1)",
          "fontFamily": "Markpro-Bold",
          "fontSize": 10,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      Menu 1
    </Text>
  </View>
  <View
    onLayout={[Function]}
    pointerEvents="box-none"
    style={
      [
        {
          "alignSelf": "center",
          "elevation": 999999,
          "flex": 0,
          "maxWidth": "100%",
          "position": "absolute",
          "zIndex": 999999,
        },
        {
          "flexDirection": "column-reverse",
          "justifyContent": "flex-start",
          "top": 30,
          "width": 750,
        },
      ]
    }
  >
    <View
      style={
        {
          "bottom": 0,
        }
      }
    >
      <RCTSafeAreaView />
    </View>
  </View>
  <View
    onLayout={[Function]}
    pointerEvents="box-none"
    style={
      [
        {
          "alignSelf": "center",
          "elevation": 999999,
          "flex": 0,
          "maxWidth": "100%",
          "position": "absolute",
          "zIndex": 999999,
        },
        {
          "bottom": 40,
          "flexDirection": "column",
          "justifyContent": "flex-end",
          "width": 750,
        },
      ]
    }
  >
    <View
      style={
        {
          "bottom": 0,
        }
      }
    >
      <RCTSafeAreaView />
    </View>
  </View>
</View>
`;
